File: app/src/main/java/www/test720/a360video/network/RxSchedulersHelper.java
package www.test720.a360video.network;

import rx.Observable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by lichuanbei on 2016/10/26.
 * 处理Rx线程:参看博文：http://www.jianshu.com/p/f3f0eccbcd6f
 */

public class RxSchedulersHelper {

    public static <T> Observable.Transformer<T, T> io_main() {
        return new Observable.Transformer<T, T>() {
            @Override
            public Observable<T> call(Observable<T> tObservable) {
                return tObservable
                        .subscribeOn(Schedulers.io())
                        .observeOn(AndroidSchedulers.mainThread());
            }
        };
    }
}

File: app/src/main/java/www/test720/a360video/network/RxSchedulersHelper.kt
package www.test720.a360video.network

import rx.Observable
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Created by lichuanbei on 2016/10/26.
 * 处理Rx线程:参看博文：http://www.jianshu.com/p/f3f0eccbcd6f
 */

object RxSchedulersHelper {

    fun <T> io_main(): Observable.Transformer<T, T> {
        return Observable.Transformer { tObservable ->
            tObservable
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
        }
    }
}
